### PR TITLE
95 serialize playerlist

### PR DESCRIPTION
**Pull request by michael-s-myer**
added serialization/deserialization to playerlist

### CODE REVIEW

File: src/playerlist.py [hunk 1]
```diff
@@ -82,3 +83,14 @@ def clear(self):
 
     def __len__(self):
         return len(self._player_list)
+
```
Comment (patmay) [inline src/playerlist.py:83]: should just be `def serialize(self)`

Comment (patmay) [reply src/playerlist.py:83]: you need to iterate through the playerlist and deliver the card_id and whether it has a corresponding player UUID

**Comment by yo09975** [comment]
Bumping back to In Progress until work is completed

**Comment by michael-s-myer** [comment]
Please review code corrections for great justice.

File: src/playerlist.py [hunk 2]
```diff
@@ -82,3 +83,15 @@ def clear(self):
...
+    def serialize(self):
+        """ Convert a Move to JSON """
+        playerlist={}
+        for player in self._player_list:
+            playerlist['card_id'] = player.get_id()
```
Comment (patmay) [inline src/playerlist.py]: Add player UUID and serialize card

resulting struct should look like:
'card_id1': {
  'card': [serialized card],
  'uuid': [player UUID or None]
}

File: src/playerlist.py [hunk 3]
```diff
@@ -82,3 +83,15 @@ def clear(self):
...
+        return json.dumps(playerlist)
+
+    def deserialize(payload):
+        """ Convert a JSON string to a Player List """
+        playerlist = json.loads(payload)
```
Comment (patmay) [inline src/playerlist.py]: you need to iterate through keys, make a player out of the Card, and assign UUID or none to the player UUID. then add that player to the playerlist

File: src/playerlist.py (hunk 2 above)
Comment (patmay) [reply src/playerlist.py]: this still needs to have the card serialized into it

**Review by patmay — APPROVED** [review]
Looks good but would like to see a sanity test

**Comment by yo09975** [comment]
Branch needs to be rebased

**Comment by yo09975** [comment]
That was fun to rebase... *tears*

**Pull request merged**